src/app/calendar/calendar.service.ts

```diff
@@ -0,0 +1,150 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import * as moment from 'moment';
+import { take, BehaviorSubject, tap, mergeMap, shareReplay, combineLatest, map, catchError, Observable, of } from 'rxjs';
+import { CalendarEvent } from './calendar-event';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CalendarService {
+
+  private eventsUrl = 'api/events';
+  httpOptions = {
+    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+  };
+
+  //fetch all user events
+  private eventsData$ = new BehaviorSubject<void>(undefined);
+  private apiRequest$ = this.http.get<CalendarEvent[]>(this.eventsUrl)
+    .pipe(
+      tap(_ => {
+        console.log('fetch events');
+      })
+    );
+  //cache events
+  private events$ = this.eventsData$.pipe(
+    mergeMap(() => this.apiRequest$),
+    shareReplay(1)
+  );
+
+  //get events in month
+  private MonthYearSubject = new BehaviorSubject<[number, number]>([parseInt(moment().format('M')), parseInt(moment().format('YYYY'))]);
+  private MonthYear$ = this.MonthYearSubject.asObservable();
+  monthEvents$ = combineLatest([
+    this.events$,
+    this.MonthYear$
+  ]).pipe(
+    tap(res => console.log(`month: ${res[1][0]}, year: ${res[1][1]}`)),
+    map(([events, date]) => {
+      let match: CalendarEvent[] = [];
+      for (let x of events) {
+        if (parseInt(moment(x.date).format('M')) === date[0] && parseInt(moment(x.date).format('YYYY')) === date[1]) {
```

Comment:
when you call moment() (or moment(x.date) in this case) more than once in a method, those two dates are slightly different and can get you into some challenging bugs down the road. Be sure to only pull `moment(x.date)` once instead of calculating it each time you need it.

Reply:
I believe that I've corrected all instances of this issue in the new commit.